aten/src/ATen/native/cuda/Blas.cpp

```diff
@@ -181,6 +181,19 @@ Tensor& addmm_out_cuda_impl(Tensor& result, const Tensor& self, const Tensor& ma
   }
 
   if (&result != &self) {
+
+    // beta*self + alpha*(mat1@mat2) = 0
+    if ((mat1._is_zerotensor() || mat2._is_zerotensor() || alpha.equal(0.0)) &&
```

Comment:
If alpha and beta are passed as zero by a user, this would surprisingly give them an efficient zero tensor. This is not a desired behavior for now since Zero Tensors are not exposed in the public API right now. I'd suggest removing the check for alpha and beta